examples/api/test/material/color_scheme/dynamic_content_color.0_test.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/color_scheme/dynamic_content_color.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // The app being tested loads images via HTTP which the test
+  // framework defeats by default.
+  setUpAll(() {
+    HttpOverrides.global = null;
+  });
+
+  testWidgets('The theme colors are created dynamically from the first image', (WidgetTester tester) async {
+    final List<(ImageProvider<Object>, Brightness)> loadColorSchemeCalls = <(ImageProvider<Object>, Brightness)>[];
+    await tester.pumpWidget(
+      example.DynamicColorExample(
+        loadColorScheme: (ImageProvider<Object> provider, Brightness brightness) async {
+          loadColorSchemeCalls.add((provider, brightness));
+          return const ColorScheme.light();
+        },
+      ),
+    );
+    await tester.pump();
```

Comment:
```suggestion

  final List<(ImageProvider<Object>, Brightness)> loadColorSchemeCalls = <(ImageProvider<Object>, Brightness)>[];

  Future<ColorScheme> fakeColorSchemeLoader(ImageProvider<Object> provider, Brightness brightness) async {
    loadColorSchemeCalls.add((provider, brightness));
    return provider == example.DynamicColorExample.images[1]
      ? ColorScheme.fromSeed(seedColor: Colors.lightBlue)
      : const ColorScheme.light();
  }

  setUp(() {
    loadColorSchemeCalls.clear();
  });

  testWidgets('The theme colors are created dynamically from the first image', (WidgetTester tester) async {
    await tester.pumpWidget(
      example.DynamicColorExample(loadColorScheme: fakeColorSchemeLoader),
    );
    await tester.pump();
```

Proposed way to share the fake loader between several tests.